modules/mcagar/src/main/ts/ephox/mcagar/api/bdd/TinyContentActions.ts

```diff
@@ -25,12 +25,40 @@ const trueClick = (editor: Editor): void =>
 const trueClickOn = (editor: Editor, selector: string): void =>
   Mouse.trueClickOn(TinyDom.body(editor), selector);
 
+const pWaitForEventToStopFiring = (editor: Editor, eventName: string, timing: { delay: number; timeout: number }): Promise<void> =>
+  new Promise((resolve, reject) => {
+    const startTime = Date.now();
+    let timer: number | undefined;
+    const onEditorEvent = () => {
+      const currentTime = Date.now();
+      if (currentTime - startTime > timing.timeout) {
+        editor.off(eventName, onEditorEvent);
+        reject(
+          `It took too long (${currentTime - startTime} ms) to stop receiving ${eventName} events. Max timeout was: ${timing.timeout} ms.`
+        );
+      } else {
+        if (timer !== undefined) {
```

Comment:
I clearTimeout of undefined is a noop. Also you could use the katamari type predicates here not sure it adds much value though.

Reply:
There are some tests that check scrolling and flipping, but nothing is all that precise. It's hard to make precise tests about things like this without just potentially making the same mistakes / assumptions in the testing code.